lib/idmapping.c

```diff
@@ -34,20 +34,11 @@ struct map_range *get_map_ranges(int ranges, int argc, char **argv)
 		return NULL;
 	}
 
-	if (ranges != ((argc + 2) / 3)) {
+	if (ranges * 3 != argc) {
```

Comment:
yeah, another way to look at it is that:

1. usage is newuidmap pid [uid loweruid count] [uid loweruid count]
2. get_map_ranges() gets called with argc-2

so the argc passed in is precisely 3 times the number of ranges.